docs/recipe_development.rst

```diff
@@ -0,0 +1,132 @@
+
+To begin, lets first inspect the ``mbuild_fcc.py`` file, a shortened snippet is below.
+
+::
+
+    import mbuild
+
+
+    class FCC(mbuild.Compound):
+        """Create a mBuild Compound with a repeating unit of the FCC unit cell.
+
+        ... (shortened for viewability)
+
+        """
+
+        def __init__(self, lattice_spacing=None, compound_to_add=None, x=1, y=1, z=1):
+            super(FCC, self).__init__()
+
+            # ... (shortened for viewability)
+
+    if __name__ == "__main__":
+        au_fcc_lattice = FCC(lattice_spacing=0.40782,
+                             compound_to_add=mbuild.Compound(name="Au"),
+                             x=5, y=5, z=1)
+        print(au_fcc_lattice)
+
+
+
+There are two very important lines in this file that we need to know when developing a plug-in for mBuild.
+
+The first is the import statement ``import mbuild``.
+We must make sure that mbuild is installed since we are inheriting from ``mbuild.Compound``. When you decide to distribute your plug-in,
+the dependencies must be listed.
+
+The second is the name of the plug-in itself. It is considered good practice to name it the name of your ``class``.
+In this case, the name of the plug-in would be ``FCC``.
+
+The last step is to edit the ``setup.py`` file such that the plug-in can be registered under the entry_point group ``mbuild.plugins``.
+
+::
+
+    from setuptools import setup
+
+    setup(
+        name="mbuild_fcc",
+        install_requires="mbuild",
+        entry_points={
+            "mbuild.plugins":[
+                "FCC = mbuild_fcc.mbuild_fcc:FCC"
+            ]
+        },
+        py_modules=["mbuild_fcc"],
+    )
+
+This is once again, a very minimal setup file, a more thoroughly tested and developed package will have more information contained within.
+
+The important section is the ``entry_points`` argument. Here we define the entry_point group we want to register with: ``"mbuild.plugins"``.
+Finally, we tell python what name to use when accessing this plug-in, in this case we call it ``FCC`` as denoted by the name before the assignment operator ``FCC =``.
+Next, we pass the "path" from the location of the ``setup.py`` file to the python file that contains the class we want to register as a plug-in: ``mbuild_fcc.mbuild_fcc``. Then, we provide the name of the class within that python file we want to make discoverable ``:FCC``.
+
+Since the ``setup.py`` file is located in the top folder of the python project, the first ``mbuild_fcc`` is the name of the folder, and the second is the name of the python file. The colon (:) is used when accessing the class from the python file itself.
+
+
+Putting it all together
+________
+
+Finally, we have ``FCC = mbuild_fcc.mbuild_fcc:FCC``.
+
+Make sure you have mBuild installed, then run the command below in the directory where the ``setup.py`` file is located.
```

Comment:
It's not clear at this point if we expect the reader to have cloned the repo; it was introduced as a package that some unnamed developer already had, not an example the user is supposed to go through. I'm not sure which is better but I prefer the latter